src/core/geometry/qgslinestring.cpp

```diff
@@ -441,15 +441,15 @@ double QgsLineString::zAt( int index ) const
   if ( index >= 0 && index < mZ.size() )
     return mZ.at( index );
   else
-    return 0.0;
+    return std::numeric_limits<double>::quiet_NaN();
```

Comment:
Could you add this explicity to the function dox, so that callers know they must anticipate possible nan values?